packages/core/src/login/webview/vue/amazonq/backend_amazonq.ts

```diff
@@ -19,11 +19,43 @@ export class AmazonQLoginWebview extends CommonAuthWebview {
         super(AmazonQLoginWebview.sourcePath)
     }
 
-    fetchConnection(): SsoConnection | undefined {
-        if (AuthUtil.instance.isConnected() && AuthUtil.instance.conn?.type === 'sso') {
-            return AuthUtil.instance.conn
+    async fetchConnections(): Promise<SsoConnection[] | undefined> {
+        const toolkitExt = vscode.extensions.getExtension('amazonwebservices.aws-toolkit-vscode')
+        const importedApi = toolkitExt?.exports
```

Comment:
we should verify the extension is installed and return [] if so

Reply:
This is done in latest commit